src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -38,24 +39,99 @@ public SchemaProvider(MySqlConnection connection)
 			{ "Triggers", FillTriggers },
 			{ "UserPrivileges", FillUserPrivileges },
 			{ "Views", FillViews },
+			{ "Foreign Keys", FillForeignKeys },
+			{ "Indexes", FillIndexColumns },
+			{ "IndexColumns", FillIndexColumns },
 		};
+
+		m_restrictionMapping = new(StringComparer.OrdinalIgnoreCase)
+		{
+			{ nameof(FillColumns), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME", "COLUMN_NAME" } },
+			{ nameof(FillForeignKeys), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME", "COLUMN_NAME" } },
+			{ nameof(FillTableConstraints), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME", "COLUMN_NAME" } },
+			{ nameof(FillIndexColumns), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME", "COLUMN_NAME" } },
+			{ nameof(FillViews), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME" } },
+			{ nameof(FillTables), new[] { "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME" } },
+			{ nameof(FillDatabases), new[] { "SCHEMA_NAME" } },
+
+			// propertly some restrictions are missing
+		};
+	}
+
+	private async Task FillForeignKeys(IOBehavior ioBehavior, DataTable dataTable, string?[] restrictionValues, CancellationToken cancellationToken)
+	{
+		dataTable.Columns.AddRange(new DataColumn[]
+		{
+			new("CONSTRAINT_CATALOG", typeof(string)),
+			new("CONSTRAINT_SCHEMA", typeof(string)),
+			new("CONSTRAINT_NAME", typeof(string)),
+			new("TABLE_CATALOG", typeof(string)),
+			new("TABLE_SCHEMA", typeof(string)),
+			new("TABLE_NAME", typeof(string)),
+			new("COLUMN_NAME", typeof(string)),
```

Comment:
MySql.Data returns the following columns:

```
CONSTRAINT_CATALOG
CONSTRAINT_SCHEMA
CONSTRAINT_NAME
TABLE_CATALOG
TABLE_SCHEMA
TABLE_NAME
MATCH_OPTION
UPDATE_RULE
DELETE_RULE
referenced_table_catalog
REFERENCED_TABLE_SCHEMA
REFERENCED_TABLE_NAME
```

Should this be expanded to include those columns?